test/tests/functional/pbs_stf.py

```diff
@@ -0,0 +1,761 @@
+
+        for i in range(13, 15):
+            resv_start = now + i * 1800
+            d = self.submit_resv(resv_start, 2, resv_dur)
+            self.assertTrue(d)
+
+        resv_start = now + 15 * 1800
+        d = self.submit_resv(resv_start, 3, resv_dur)
+        self.assertTrue(d)
+
+        resv_start = now + 16 * 1800
+        d = self.submit_resv(resv_start, 2, resv_dur)
+        self.assertTrue(d)
+
+        a = {'Resource_List.max_walltime': '10:00:00',
+             'Resource_List.min_walltime': '00:10:00'}
+        j = Job(TEST_USER, attrs=a)
+
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        attr = {'Resource_List.walltime': (LE, '04:00:00')}
+        self.server.expect(JOB, attr, id=jid)
+
+        attr = {'Resource_List.walltime': (GE, '00:10:00')}
+        self.server.expect(JOB, attr, id=jid)
+
+    def test_t_4_3_6(self):
+        """
+        Test shrink to fit by creating one reservation having ncpus=1,
+        starting in 3 min. with a duration of two hours.  A preempted STF job
+        with min_walltime of 2 min. and max_walltime of 2 hours will stay
+        suspended after higher priority job goes away if its
+        min_walltime can't be satisfied.
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        qname = 'highp'
+
+        queues = self.server.status(QUEUE)
+        a = {'queue_type': 'execution'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, qname)
+        
+        a = {'enabled': 'True', 'started': 'True', 'priority': '150'}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, qname)
+
+        now = int(time.time())
+        resv_dur = 7200
+
+        resv_start = now + 180
+        d = self.submit_resv(resv_start, 1, resv_dur)
+        self.assertTrue(d)
+
+        a = {'Resource_List.max_walltime': '02:00:00',
+             'Resource_List.min_walltime': '00:02:00'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        attr = {'Resource_List.walltime': (LE, '00:03:00')}
+        self.server.expect(JOB, attr, id=jid)
+
+        attr = {'Resource_List.walltime': (GE, '00:02:00')}
+        self.server.expect(JOB, attr, id=jid)
+
+        # The sleep below will leave less than 2 minutes window for jid
+        # after j2id is deleted.  The min_walltime of jid can't be
+        # satisfied and jid will stay in S state.
+        self.logger.info("Sleeping 65s to leave less than 2m before resv")
+        time.sleep(65)
+
+        a = {'queue': 'highp', 'Resource_List.select': '1:ncpus=1',
+             'Resource_List.walltime': '00:01:00'}
+        j = Job(TEST_USER, attrs=a)
+        j2id = self.server.submit(j)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=j2id)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid)
+
+        time.sleep(5)
```

Comment:
do you know why are we sleeping here?

Reply:
I added this sleep to allow the preempted job to run for a little while.  After reviewing it, the initial 65s sleep should handle what the test case requires.  This test fails due to PP-1049.  I am now skipping it.